src/components/shapediver/appbuilder/AppBuilderDrawingToolsWidgetComponent.tsx

```diff
@@ -0,0 +1,214 @@
+					}
+				},
+				{
+					definition: {
+						id: PARAMETER_NAMES.GRID_UNIT,
+						name: "Grid Unit",
+						defval: (planeRestrictionApi?.gridRestrictionApi.gridUnit ?? 1) + "",
+						min: 1,
+						max: 10,
+						type: ShapeDiverResponseParameterType.INT,
+						hidden: drawingToolsActive === false || planeRestrictionApi?.gridRestrictionApi.gridUnitEditable === false
+					}
+				},
+				{
+					definition: {
+						id: PARAMETER_NAMES.ANGULAR_SNAPPING,
+						name: "Angular Snapping",
+						defval: (planeRestrictionApi?.angularRestrictionApi.enabled ?? false) + "",
+						type: ShapeDiverResponseParameterType.BOOL,
+						hidden: drawingToolsActive === false || planeRestrictionApi?.angularRestrictionApi.enabledEditable === false
+					}
+				},
+				{
+					definition: {
+						id: PARAMETER_NAMES.ANGULAR_SECTIONS,
+						name: "Angular Sections",
+						defval: Math.PI / (planeRestrictionApi?.angularRestrictionApi.angleStep ?? Math.PI / 8) + "",
+						min: 2,
+						max: 24,
+						type: ShapeDiverResponseParameterType.INT,
+						hidden: drawingToolsActive === false || planeRestrictionApi?.angularRestrictionApi.angleStepEditable === false
+					}
+				}
+			]
+		);
+	}, [drawingToolsApi, drawingToolsActive]);
+
+
+	// define the custom drawing tools parameters and a handler for changes
+	const customSessionId = "mysession";
+	useDefineGenericParameters(customSessionId, false /* acceptRejectMode */,
+		materialParameters,
+		async (values) => {
+			if (PARAMETER_NAMES.START_DRAWING in values)
+				setDrawingToolsActive("" + values[PARAMETER_NAMES.START_DRAWING] === "true");
+
+			if (drawingToolsApi !== undefined) {
+				const planeRestrictionApi = Object.values(drawingToolsApi.restrictions).find(restriction => restriction instanceof PlaneRestrictionApi)! as PlaneRestrictionApi;
+
+				if (PARAMETER_NAMES.SHOW_POINT_LABELS in values)
+					drawingToolsApi.showPointLabels = "" + values[PARAMETER_NAMES.SHOW_POINT_LABELS] === "true";
+
+				if (PARAMETER_NAMES.SHOW_DISTANCE_LABELS in values)
+					drawingToolsApi.showDistanceLabels = "" + values[PARAMETER_NAMES.SHOW_DISTANCE_LABELS] === "true";
+
+				if (PARAMETER_NAMES.GRID_SNAPPING in values)
+					planeRestrictionApi.gridRestrictionApi.enabled = "" + values[PARAMETER_NAMES.GRID_SNAPPING] === "true";
+
+				if (PARAMETER_NAMES.GRID_UNIT in values)
+					planeRestrictionApi.gridRestrictionApi.gridUnit = values[PARAMETER_NAMES.GRID_UNIT];
+
+				if (PARAMETER_NAMES.ANGULAR_SNAPPING in values)
+					planeRestrictionApi.angularRestrictionApi.enabled = "" + values[PARAMETER_NAMES.ANGULAR_SNAPPING] === "true";
+
+				if (PARAMETER_NAMES.ANGULAR_SECTIONS in values)
+					planeRestrictionApi.angularRestrictionApi.angleStep = Math.PI / values[PARAMETER_NAMES.ANGULAR_SECTIONS];
+			}
+
+			return values;
+		}
+	);
+	const myParameterProps = useSessionPropsParameter(customSessionId);
+
+	if (parameterName !== undefined)
+		return <ParametersAndExportsAccordionComponent key={0}
```

Comment:
@MajorMeerkatThe3rd I think the key here would better be implemented using a unique id which we generate only once, e.g. using `useState` with an initializer.

Reply:
@snabela I used the mantine function for that, should that be the same in the other files where the key={0} is used? (AppBuilderFallbackContainerComponent, AppBuilderTabsComponent, ModelSelect)